estimagic/logging/database_utilities.py

```diff
@@ -287,6 +298,33 @@ def read_last_rows(
     return _execute_read_statement(database, table_name, stmt, return_type)
 
 
+def read_specific_row(
+    database, table_name, rowid, return_type, path=None, fast_logging=False
+):
+    """Read a specific row from a table.
+
+    If this row does not exist, a KeyError is raised.
```

Comment:
This function does not raise a `KeyError` but returns a dict of empty lists (in case of `return_type="dict_of_lists"`). The KeyError is raised in `read_optimization_iteration`.

Reply:
Done